src/Observable.ts

```diff
@@ -111,7 +113,11 @@ function triggerListeners(obj, newObject) {
     //     triggerListeners(val, newObject[key]);
     //   });
     // } else {
-    triggerListeners(obj[key], newObject[key]);
+
+    // this is when we are upadting an array with less values than the orignial
+    if (obj && newObject) {
```

Comment:
with the above comment, the newObject was null since it contains less items than the original. This will also support the inverse (obj being null(